src/js/reducers/app-state-reducer.js

```diff
@@ -0,0 +1,75 @@
+import { ENABLE_SETTINGS, DISABLE_SETTINGS, TOGGLE_WIDGET, OPEN_WIDGET, CLOSE_WIDGET, SHOW_SETTINGS, HIDE_SETTINGS, SHOW_SETTINGS_NOTIFICATION, HIDE_SETTINGS_NOTIFICATION, SET_SERVER_URL, SET_EMAIL_READONLY, UNSET_EMAIL_READONLY } from 'actions/app-state-actions';
+import { RESET } from 'actions/common-actions';
+
+const INITIAL_STATE = {
+    settingsVisible: false,
+    settingsNotificationVisible: false,
+    widgetOpened: undefined,
+    settingsEnabled: true,
+    readOnlyEmail: false,
+    serverURL: 'https://api.smooch.io/'
+};
+
+export function AppStateReducer(state = INITIAL_STATE, action) {
+    switch (action.type) {
+        case RESET:
+            return Object.assign({}, INITIAL_STATE);
+
+        case ENABLE_SETTINGS:
+            return Object.assign({}, state, {
```

Comment:
I agree there is duplication, but IMO it's easier to read than splitting the logic in two different places instead of having everything in the same switch statement.